### PR TITLE
CLDR-14336 for currency patterns add alts alphaNextToNumber,noCurrency; add element currencyPatternAppendISO

### PR DESCRIPTION
**Pull request by pedberg-icu**
CLDR-14336

- [ ] This PR completes the ticket.

 
This does the following:
-  For currency patterns (standard, accounting, and compact decimal) adds two alt= forms: alphaNextToNumber (for when the currency symbol character nearest to the number is an alpha character) and noCurrency (to get a currency-style pattern without the actual currency symbol). These are mutually exclusive.
    - The alphaNextToNumber form is generally only needed to add space between the symbol and number if the pattern does not already have a space there.
    - For compact decimals, the noCurrency form is only needed if the result needs to be different from the short compact decimal formats, which provide the fallback for these. Currently there are no such cases (formerly the `pt` compact currency formats would have needed this). However, the noCurrency form is still useful for the standard & accounting currency formats (e.g. to get the format with 2 trailing decimals but without a currency sign).
- Under currencyFormats adds a new child element currencyPatternAppendISO which provides a message pattern intended for combining a currency format that uses a symbol with the ISO 4217 code for the currency (e.g. to get $1,432.00 USD). In this PR that pattern exists only in root.
- Adds the necessary data for alphaNextToNumber and noCurrency to locales in common/main/. The alphaNextToNumber patterns are based on the current currencySpacing behavior. All of this data is draft="provisionsal", except for `en`.
- Updates tests, coverage, PathHeader, PathDescription, etc.

Also in SandboxLocalesTest I had to change the minimalDraftStatus for Factory.make from null to unconfirmed, otherwise there was an NPE in XMLNormalizingLoader for locale "root" in the call to cache.getUnchecked(key).

### CODE REVIEW

File: tools/cldr-apps/src/test/java/org/unicode/cldr/util/SandboxLocalesTest.java [hunk 1]
```diff
@@ -45,6 +45,8 @@ void test() throws IOException {
         // our factory includes common/main, so limit here.
         for(final CLDRLocale l : SpecialLocales.getByType(SpecialLocales.Type.scratch)) {
             System.out.println("Testing " + l);
+            // TODO With the current changes for CLDR-14336 there is an NPE in the following,
+            // see further notes in SimpleFactory.handleMake
```
Comment (pedberg-icu) [inline tools/cldr-apps/src/test/java/org/unicode/cldr/util/SandboxLocalesTest.java:49]: To investigate

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/SimpleFactory.java [hunk 2]
```diff
@@ -524,6 +524,11 @@ public CLDRFile handleMake(String localeName, boolean resolved, DraftStatus mini
                 try {
                     makeResolvingSource = makeResolvingSource(localeName, minimalDraftStatus);
                 } catch (Exception e) {
+                    // TODO: The current changes for CLDR-14336 result in a NPE being caught below when this
+                    // is called from SandboxLocalesTest.test(). In that case this handleMake method is called
+                    // recursively via Factory.makeResolvingSource; the third time it is called is with
+                    // localeName=root and resolved=false, and for that case the call to new CLDRFile below
+                    // generates the NPE.
```
Comment (pedberg-icu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/SimpleFactory.java:531]: The focus of the failure investigation

File: tools/cldr-apps/src/test/java/org/unicode/cldr/util/SandboxLocalesTest.java [hunk 3]
```diff
@@ -45,7 +46,14 @@ void test() throws IOException {
         // our factory includes common/main, so limit here.
         for(final CLDRLocale l : SpecialLocales.getByType(SpecialLocales.Type.scratch)) {
             System.out.println("Testing " + l);
-            CLDRFile f = factory.make(l.getBaseName(), true, null);
```
Comment (srl295) [inline tools/cldr-apps/src/test/java/org/unicode/cldr/util/SandboxLocalesTest.java:48]: Could you just omit the third argument? Then it uses the factory's default.  This was definitely a mistake, I didn't have a reason to pass null here.

**Review by btangmu — APPROVED** [review]

File: tools/cldr-apps/src/test/java/org/unicode/cldr/util/SandboxLocalesTest.java (hunk 3 above)
Comment (pedberg-icu) [reply tools/cldr-apps/src/test/java/org/unicode/cldr/util/SandboxLocalesTest.java:48]: I will try that in a future cleanup, I would like to get this merged so we can start shakedown.

Comment (pedberg-icu) [reply tools/cldr-apps/src/test/java/org/unicode/cldr/util/SandboxLocalesTest.java:48]: I did that in https://github.com/unicode-org/cldr/pull/2009

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Placeholders.txt [hunk 4]
```diff
@@ -203,6 +203,8 @@
 
 ^//ldml/numbers/currencyFormats/unitPattern ; {0}=NUMBER 3; {1}=CURRENCY_NAME Dollars
 ^//ldml/numbers/currencyFormats\[@numberSystem="%A"]/unitPattern ; {0}=NUMBER 3; {1}=CURRENCY_NAME Dollars
+^//ldml/numbers/currencyFormats/currencyPatternAppendISO ; {0}=CURRENCY_FORMAT ¤#,##0.00
```
Comment (sffc) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Placeholders.txt:206]: I don't see any seed data for `currencyPatternAppendISO`; is that intentional?

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java [hunk 5]
```diff
@@ -138,12 +138,14 @@ public void testCurrency() {
         "//ldml/numbers/currencyFormats/currencySpacing/afterCurrency/currencyMatch",
         "//ldml/numbers/currencyFormats/currencySpacing/afterCurrency/surroundingMatch",
         "//ldml/numbers/currencyFormats/currencySpacing/afterCurrency/insertBetween",
+        "//ldml/numbers/currencyFormats/currencyPatternAppendISO", // TODO see CLDR-14831
```
Comment (sffc) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java:141]: CLDR-14831 doesn't mention anything about currency formatting.

**Review by sffc — COMMENTED** [review]
Thank you for doing this! I left a couple questions.

(This is my long-delayed Jira ticket review)

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java (hunk 5 above)
Comment (pedberg-icu) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java:141]: Yes, that is in the linked ticket https://unicode-org.atlassian.net/browse/CLDR-15286

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Placeholders.txt (hunk 4 above)
Comment (pedberg-icu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Placeholders.txt:206]: There is a pattern in root. Locales can override it if they choose to.

**Pull request merged**